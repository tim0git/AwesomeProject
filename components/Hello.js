import React from 'react';
// import PropTypes from 'prop-types';
import {View, Text} from 'react-native';

function Hello() {
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text id="welcome-message">Hello World</Text>
    </View>
  );
}

// Hello.propTypes = {};

export default Hello;
