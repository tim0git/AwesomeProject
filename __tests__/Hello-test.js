import 'react-native';
import React from 'react';
import 'jest-enzyme';
import 'react-native-mock-render/mock';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

// import {shallow, mount} from 'enzyme';
// import sinon from 'sinon';
// import renderer from 'react-test-renderer';

import Hello from '../components/Hello';

describe('<Hello >', () => {
  // jest.useFakeTimers();
  const hello = <Hello />;

  it('Component Renders', () => {
    const component = shallow(hello);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('has text hello', () => {
    const wrapper = shallow(hello);
    expect(
      wrapper
        .find('Text')
        .at(0)
        .findWhere((w) => w.text() === 'Hello World'),
    ).toExist();
  });
});
