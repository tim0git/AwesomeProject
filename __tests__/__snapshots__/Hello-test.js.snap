// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hello > Component Renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    id="welcome-message"
  >
    Hello World
  </Text>
</View>
`;
